contracts/actions/compoundV3/CompV3Transfer.sol

```diff
@@ -0,0 +1,83 @@
+import "../../utils/TokenUtils.sol";
+import "../ActionBase.sol";
+import "./helpers/CompV3Helper.sol";
+
+/// @title Payback a token a user borrowed from Compound
+contract CompV3Transfer is ActionBase, CompV3Helper {
+    using TokenUtils for address;
+
+    struct Params {
+        address from;
+        address to;
+        address asset;
+        uint256 amount;
+    }
+    error CompV3TransferError();
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes memory _callData,
+        bytes32[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+
+        params.from = _parseParamAddr(params.from, _paramMapping[0], _subData, _returnValues);
+        params.to = _parseParamAddr(params.to, _paramMapping[1], _subData, _returnValues);
+        params.asset = _parseParamAddr(params.asset, _paramMapping[2], _subData, _returnValues);
+        params.amount = _parseParamUint(params.amount, _paramMapping[3], _subData, _returnValues);
+
+        (uint256 withdrawAmount, bytes memory logData) = _transfer(params.from, params.to, params.asset, params.amount);
+        emit ActionEvent("CompV3Transfer", logData);
+        return bytes32(withdrawAmount);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes memory _callData) public payable override {
+        Params memory params = parseInputs(_callData);
+        (, bytes memory logData) = _transfer(params.from, params.to, params.asset, params.amount);
+        logger.logActionDirectEvent("CompV3Transfer", logData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Transfer amount of supplied assets from one address to another
+    /// @dev Amount type(uint).max will transfer the whole amount of supplied assets
+    /// @param _from The address of an account that is the sender of the asset in the transaction
+    /// @param _to The address of an account that is the receiver in the transaction
+    /// @param _asset The ERC-20 address of the asset that is being sent in the transaction
+    /// @param _amount Amount of the specified asset to be transfered
+    function _transfer(
+        address _from,
+        address _to,
+        address _asset,
+        uint256 _amount
+    ) internal returns (uint256, bytes memory) {
+        address tokenAddr = _asset;
+
+        if (_from != address(0)) {  //sends assets from one EOA to another
+            IComet(COMET_ADDR).transferAssetFrom(_from, _to, tokenAddr, _amount);
+        }
+        else {                      //sends assets from this address to dst
+            IComet(COMET_ADDR).transferAsset(_to, tokenAddr, _amount);
```

Comment:
I'd maybe just use 
IComet(COMET_ADDR).transferAssetFrom(_from, _to, tokenAddr, _amount);
in both cases and would add some checks
if (_from == address(0)) _from = address(this)
require(_to != address(0))